python/lib/repository_standards.py

```diff
@@ -0,0 +1,201 @@
+"""
+This module contains the classes used to generate the GitHub repository standards report.
+
+- The OrganisationStandardsReport class is used to generate a report for a given organisation.
+- The RepositoryStandards class is used to generate a report for a given repository.
+
+The report is generated to publish which repositories are compliant with the standards set out by the
+Operations Engineering team. The report is sent to the operations-engineering-reports API.
+"""
+
+import json
+
+import requests
+from cryptography.fernet import Fernet
+
+
+class OrganisationStandardsReport:
+    """
+    This class is used to generate a report for a given organisation.
+    The most important aspect of this class is the report collection, which should contain a list of
+    RepositoryStandards objects.
+    """
+
+    def __init__(self, endpoint: str, api_key: str, enc_key: hex) -> None:
+        self.report: list = []
+        self.api_endpoint = endpoint
+        self.api_key = api_key
+        self.encryption_key = enc_key
+
+    def all_reports(self) -> list:
+        """
+        Return the collection of reports.
+        """
+        return self.report
+
+    def add(self, report) -> None:
+        """
+        Add a RepositoryStandards object to the collection of reports.
+        """
+        self.report.append(report)
+
+    def send_to_api(self) -> None:
```

Comment:
Think this could be seperated into it's own Service Class? I.e. `operations_engineering_service`?

Would mean this class can remove all the HTTP stuff and just represent the domain model

Reply:
Refactored.